src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -242,14 +246,27 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
             num_class = train_cfg.get("num_class", None)
             objective = train_cfg.get("objective", None)
             # RepeatedStratifiedKFold expects X as array-like of shape (n_samples, n_features)
-            X = range(train_val_dmatrix.num_row())
-            y = train_val_dmatrix.get_label() if num_class or objective.startswith("binary:") else None
+            classification_problem = num_class or objective.startswith("binary:")
+            num_rows_in_dataset = train_val_dmatrix.num_row()
+            X = range(num_rows_in_dataset)
+            y = train_val_dmatrix.get_label() if classification_problem else None
             rkf = RepeatedStratifiedKFold(n_splits=kfold, n_repeats=num_cv_round) if y is not None \
                 else RepeatedKFold(n_splits=kfold, n_repeats=num_cv_round)
 
-            for train_index, val_index in rkf.split(X=X, y=y):
+            # For regression, ground truth and predicted values are stored. For classification, additionally
```

Comment:
Can we call out here in the comments that the next block is for initializing variables needed for storing postprocessing data. This train_job method is already large and needs to be broken up. For now let's be clear here to make it easier in the future to separate things out. I want to scope out what's needed for particular parts of this method.

Reply:
I agree. 

I am thinking to have a class that is responsible for aggregation of predictions, and storing those predictions to file. This way it is easier to extend to e.g. make predictions on non cv folds, easier to copy to other algorithm containers, and most of the new code would be factored in.